src/ars/core/repository.py

```diff
@@ -307,9 +307,32 @@ async def register_dataset(self, dataset: Dataset) -> None:
     async def delete_dataset(self, dataset_id: str) -> None:
         """Remove the dataset with the given ID.
 
+        All pending access requests pertaining to the dataset ID are set to denied.
         Raises a `DatasetNotFoundError` if the dataset was not found.
         """
         try:
+            async for request in self._request_dao.find_all(
+                mapping={"dataset_id": dataset_id}
+            ):
+                if request.status == AccessRequestStatus.PENDING:
+                    update = {
+                        "status": AccessRequestStatus.DENIED,
+                        "status_changed": now_as_utc(),
+                        "note_to_requester": "This dataset has been deleted",
+                        # should this be a value to indicate this has happend automatically instead?
+                        "changed_by": None,
+                    }
+                    updated_request = request.model_copy(update=update)
+                    await self._request_dao.update(updated_request)
+                elif (
+                    request.status == AccessRequestStatus.ALLOWED
+                    and request.access_ends > now_as_utc()
+                ):
+                    log.warning(
+                        "A valid access request with ID %s still exists for the deleted dataset with ID %s.",
+                        request.id,
+                        dataset_id,
+                    )
             return await self._dataset_dao.delete(id_=dataset_id)
```

Comment:
Wouldn't it be better to first do the deletion, and then adapt the access requests? This also raises an error if the dataset does not exist, so we wouldn't need to continue.

Reply:
Yep, let me quickly change that.
Seems I'm having a severe case of Monday brain today...